Avoid having subdirectories for non-go files

keisar opened the pull request:
As per cgo's documentation all non-go files should reside in the same folder as the go files and not in a subdirectory. Until now this has not been an issue because most vendoring tools allow keeping "/include" but it does fails with `go mod vendor` (vendor directory doesn't include the header files).

In this PR:

- Moved all header files to the parent directory
- Updated all header files to have the "h3_" prefix
- Updated the update-h3.sh script to add the prefix to the header files
- Since we're changing the name of the header files I also updated the udpate-h3.sh script to alter the files to use the new name convention (using sed)

Important note: The update-h3.sh only works partially, there are two C files (h3_localij.c & h3_h3Index.c) which include header files twice, once with <> and another with "", for now it requires manually removing the include statement that uses <>.
I believe this should be addressed in the Core Library because there's no need to include the header twice, if others agree as well I can create a PR for that in the Core Library.

Related issue: https://github.com/uber/h3-go/issues/30

isaacbrodsky commented on .gitignore:
```diff
@@ -26,3 +26,9 @@ _testmain.go
+# Workspace files
+.idea
+
+# Intellij project files
+.idea
```
Duplicate gitignore?

keisar commented:
@joegilley @isaacbrodsky I removed the duplication and also removed some forgotten header files.

Created the PR for the core library here: https://github.com/uber/h3/pull/333/files

keisar commented:
Hi @joegilley thanks for the approval, do you know approx. when it can be merged and released in a new version?
No pressure from my end as we're using a fork for now.

joegilley commented:
@keisar thanks for patience! I’ll push a new version with your changes tomorrow!
